sydent/http/servlets/store_invite_servlet.py

```diff
@@ -147,8 +147,26 @@ def redact_email_address(self, address):
         # Extract strings from the address
         username, domain = address.split(u"@", 1)
 
-        # Obfuscate strings
-        redacted_username = self._redact(username, self.sydent.username_obfuscate_characters)
+        # Obfuscate the username portion
+
+        # Break up the username by the configured separator string
+        separator = self.sydent.third_party_invite_username_separator_string
+        username_components = username.split(separator) if separator else [username]
+
+        # Obfuscate each individual component
+        redacted_components = [
+            self._redact(component, self.sydent.username_obfuscate_characters)
+            for component in username_components if component != ""
+        ]
```

Comment:
I don't know if it makes sense to drop the empty portions, if you have `donald--fauntleroy--duck@disney.com` I think I'd expect this (with a count of 3) to become `don...--fau...--duc...`, not `don...-fau...-duc`.

I think this also guarantees that `redacted_components` will be truthy also?

Reply:
> I think this also guarantees that redacted_components will be truthy also?

Well, if the username == the separator then `redacted_components` would be an empty list, and thus `False`.

Though I'm likely going to remove `if component != ""` based on other comments.

Reply:
Well, removing `if component != ""` has the side-effect of turning:

`applejack-@someexample.com` into `applej...-...@someexample.com`

Which may be a fine compromise?